packages/angular-cli/utilities/check-package-manager.ts

```diff
@@ -0,0 +1,60 @@
+import * as chalk from 'chalk';
+import {exec} from 'child_process';
+import {CliConfig} from '../models/config';
+
+const Promise = require('../ember-cli/lib/ext/promise');
+const config = CliConfig.fromProject();
+
+let packageManager = 'npm';
+if (config && config.get('packageManager')) {
+  packageManager = config.get('packageManager');
+}
+
+export function checkYarnOrCNPM() {
+  return new Promise((resolve, reject) => {
+    Promise
+      .all([checkYarn(), checkCNPM()])
+      .then((data) => {
+        if (packageManager === 'npm') {
```

Comment:
If this checks is done by the function calling this, it would reduce the code duplication and make the overall structure easier to understand.